lms/djangoapps/courseware/tests/test_entrance_exam.py

```diff
@@ -206,6 +228,186 @@ def setUp(self):
             ]
         )
 
+    @mock.patch('xmodule.x_module.XModuleMixin.has_dynamic_children', mock.Mock(return_value='True'))
+    def test_view_redirect_if_entrance_exam_required(self):
+        """
+        Unit Test: if entrance exam is required. Should return a redirect.
+        """
+        url = reverse('courseware', kwargs={'course_id': unicode(self.course.id)})
+        expected_url = reverse('courseware_section',
+                               kwargs={
+                                   'course_id': unicode(self.course.id),
+                                   'chapter': self.entrance_exam.location.name,
+                                   'section': self.exam_1.location.name
+                               })
+        resp = self.client.get(url)
+        self.assertRedirects(resp, expected_url, status_code=302, target_status_code=200)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENTRANCE_EXAMS': False})
+    def test_entrance_exam_content_absence(self):
+        """
+        Unit Test: If entrance exam is not enabled then page should be redirected with chapter contents.
+        """
+        url = reverse('courseware', kwargs={'course_id': unicode(self.course.id)})
+        expected_url = reverse('courseware_section',
+                               kwargs={
+                                   'course_id': unicode(self.course.id),
+                                   'chapter': self.chapter.location.name,
+                                   'section': self.chapter_subsection.location.name
+                               })
+        resp = self.client.get(url)
+        self.assertRedirects(resp, expected_url, status_code=302, target_status_code=200)
+        resp = self.client.get(expected_url)
+        self.assertNotIn('Exam Problem - Problem 1', resp.content)
+        self.assertNotIn('Exam Problem - Problem 2', resp.content)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENTRANCE_EXAMS': True})
+    def test_entrance_exam_content_presence(self):
+        """
+        Unit Test: If entrance exam is enabled then its content e.g. problems should be loaded and redirection will
+        occur with entrance exam contents.
+        """
+        url = reverse('courseware', kwargs={'course_id': unicode(self.course.id)})
+        expected_url = reverse('courseware_section',
+                               kwargs={
+                                   'course_id': unicode(self.course.id),
+                                   'chapter': self.entrance_exam.location.name,
+                                   'section': self.exam_1.location.name
+                               })
+        resp = self.client.get(url)
+        self.assertRedirects(resp, expected_url, status_code=302, target_status_code=200)
+        resp = self.client.get(expected_url)
```

Comment:
You can just add these 3 lines (279-281) to test_view_redirect_if_entrance_exam_required and delete this test case. I believe it is redundant with test_view_redirect_if_entrance_exam_required, though I don't understand why test_view_redirect_if_entrance_exam_required requires a mock of xmodule.x_module.XModuleMixin.has_dynamic_children and also does not need to specify {'ENTRANCE_EXAMS': True}.

Please just take a look at these 2 test cases and see if it makes sense to consolidate them.